### ISSUE TITLE
batchnorm does not implement Module to be included in a sequential model

### ISSUE THREAD

**Issue by OfekShochat** [LaurentMazare/tch-rs#430]
I wanted to include a batchnorm inside my model and I wondered if I could do this within a sequential model. when I try to `seq.add(nn::batch_norm2d(vs / "bn1", filters, nn::BatchNormConfig::default()))` it outputs: ```the trait `Module` is not implemented for `BatchNorm` ```. what to do?
thanks aot

**Comment by OfekShochat** [comment]
so Ill not close this issue (yet) but I just made a structure and implemented Module for it. Im making a residual network that is so I just needed it to make the residual block so it worked out perfectly

**Comment by LaurentMazare** [comment]
The reason why `BatchNorm` doesn't implement `Module` is that the layer has to know whether it's in training mode or inference mode, hence it implements `ModuleT` instead.
As this issue had no activity in a while, I'm closing it now but feel free to re-open if it's still a problem.
